### DISCUSSION TITLE
Server rejected WebSocket connection

### DISCUSSION THREAD

**Discussion by WTIgraysonm** [deepgram/community#822] [General help] [answered]
Error:
> WebSocketException in LiveClient.start: server rejected WebSocket connection: HTTP 400
> ERROR:deepgram.clients.live.v1.client:WebSocketException in LiveClient.start: server rejected WebSocket connection: HTTP 400

What I'm attempting to do:
I'm setting up a WebSocket using FastAPI. This is to accept audio data as byte stream arrays, and pass it on to Deepgram to transcribe. The audio data is fixed at a specific Rate, Chunk size, and mulaw audio type. And it is multi channel with 2 different audio sources on the individual channels (speaker 1 and speaker 2).

I'm using the python Deepgram-SDK to setup a live transcription.

My Deepgram Code to setup the initial connection:
```
async def setup_deegram(data):
    try:
        config = DeepgramClientOptions(
            #options={"keepalive": "true"} # Comment this out to see the effect of not using keepalive
        )
        
        deepgram = DeepgramClient(DEEPGRAM_API_KEY, config)

        dg_connection = deepgram.listen.live.v("1")

        def on_message(self, result, **kwargs):
            channel_index = result.channel_index[0]
            sentence = result.channel.alternatives[0].transcript
            if len(sentence) == 0:
                return
            if channel_index == 0:
                print(f"User: {sentence}")
            else:
                print(f"IVR: {sentence}")

        def on_error(self, error, **kwargs):
            print(f"\n\n{error}\n\n")

        dg_connection.on(LiveTranscriptionEvents.Transcript, on_message)
        dg_connection.on(LiveTranscriptionEvents.Error, on_error)

        options = LiveOptions(
            model="nova-2", 
            language="en-US", 
            smart_format=True,
            multichannel=True,
            channels=2,
            sample_rate=RATE,
            encoding="mulaw",
            vad_events=True,
            utterance_end_ms="1500",
        )
        
        dg_connection.start(options)
        
        return dg_connection

    except Exception as e:
        logger.error(f"Could not open socket: {e}")
```

Now what's odd is that this was working correctly the other day. I cleaned up some dependencies and re-built my environment and now it's not working. Nothing related to Deepgram was changed. Here are my current dependencies:
```
python = ">=3.10.0,<3.12"
numpy = "^1.22.2"
Flask = "^2.2.0"
pydantic = "^2.0"
python-dotenv = "^1.0.1"
deepgram-sdk = "^3.3.1"
passlib = "^1.7.4"
bcrypt = "^4.1.3"
uvicorn = "^0.30.1"
fastapi = "^0.111.0"
```

As far as I can tell Deepgram is setup correctly, but I'm not sure what is causing the error.

Now I know there should be a deepgram request ID attached to my attempts, but I don't know how to acquire it via the SDK. So I apologize for not providing it. If you can help me get that ID I'm sure that would help getting the Deepgram logs.
Also if there's a way to get a more detailed error back from Deepgram other than what I've got that would also help narrow down the issue.

I've tried playing with the LiveOptions, but they haven't made a difference. I've also regenerated my API Key to make sure that was valid, and no luck.

Please let me know if there's anything else I can provide.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by WTIgraysonm** [comment, marked as answer]
After playing with it for a while I figured out a few things.

The cause of the error was:
`utterance_end_ms="1500",`

This was working yesterday, and today it's not. Thankfully I don't need it and can have it commented out. Now it runs fine.

Another discovery was the `on_metadata()` is what should return the request_id. But since this was failing before that could be run (and isn't in my original code) I never got one to post here.
Here's the piece of code to add:
```
        def on_metadata(self, metadata, **kwargs):
            logger.debug(f"\nMetadata:\n{metadata}\n\n")
        dg_connection.on(LiveTranscriptionEvents.Metadata, on_metadata)
```
